Remove /scripts/build-consumer and documentation (#5958)
Shopify/polaris: polaris-react/README.md

### Testing in a consuming project

> The `/snapit` GitHub comment command in pull requests will publish a snapshot NPM package for testing.

Reviewer: **suggestion:** Should we add a link to the [snapshot-release docs](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#snapshot-release)?

> The `/snapit` GitHub comment command in pull requests will publish a snapshot NPM package for testing.

Author: ```suggestion
The `/snapit` GitHub comment command in pull requests will publish a snapshot NPM package for testing. Read the [release documentation](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#snapshot-release) for more information.
```

> To test your changes in a consuming product you can run the `/snapit` command in your pull request to publish a snapshot NPM package for testing.

Author: ```suggestion
The `/snapit` GitHub comment command in pull requests will publish a snapshot NPM package for testing.
```
